yast/lib/suse/connect/package_search.rb

```diff
@@ -0,0 +1,30 @@
+require 'suse/toolkit/shim_utils'
+
+module SUSE
+  module Connect
+    # Enable connect and zypper extensions/scripts to search packages for a
+    # certain product
+    class PackageSearch
+      class << self
+        include SUSE::Toolkit::ShimUtils
+
+        # Search packages depending on the product and its extension/module
+        # tree.
+        #
+        # @param query [String] package to search
+        # @param product [SUSE::Connect::Zypper::Product] product to base search on
+        # @param config_params [<Hash>] overwrites from the config file
+        #
+        # @return [Array< <Hash>>] Returns all matched packages or an empty array if no matches where found
+        def search(query, product:, config_params: {})
```

Comment:
How do I invoke this from Yast? I tried Search from `Software->Software Management`, but I don't see the Debug Merged message and it seems to be doing a local search.

Reply:
`Software -> Software Management -> Extras -> Search Online`
See: https://github.com/yast/yast-registration/pull/467

Note that you might not see some of the debug messages in YaST log if they are printed before the loggers are redirected.

Reply:
I moved the debug code around a bit. You should see most of the early messages if you have debug enabled.

Reply:
okay, my terminal was too small and the menu with Extras was not shown